nearcore/src/entity_debug.rs

```diff
@@ -0,0 +1,848 @@
+                    .ok_or_else(|| anyhow!("Block header not found"))?;
+                Ok(serialize_entity(&BlockHeaderView::from(block_header)))
+            }
+            EntityQuery::BlockHashByHeight { block_height } => {
+                let block_hash = self
+                    .store
+                    .get_ser::<CryptoHash>(DBCol::BlockHeight, &block_height.try_to_vec().unwrap())?
+                    .ok_or_else(|| anyhow!("Block height not found"))?;
+                Ok(serialize_entity(&block_hash))
+            }
+            EntityQuery::ChunkByHash { chunk_hash } => {
+                let chunk = self
+                    .store
+                    .get_ser::<ShardChunk>(DBCol::Chunks, &chunk_hash.try_to_vec().unwrap())?
+                    .ok_or_else(|| anyhow!("Chunk not found"))?;
+                let epoch_id =
+                    self.epoch_manager.get_epoch_id_from_prev_block(chunk.prev_block())?;
+                let author = self.epoch_manager.get_chunk_producer(
+                    &epoch_id,
+                    chunk.height_created(),
+                    chunk.shard_id(),
+                )?;
+                Ok(serialize_entity(&ChunkView::from_author_chunk(author, chunk)))
+            }
+            EntityQuery::EpochInfoByEpochId { epoch_id } => {
+                let epoch_info = self.epoch_manager.get_epoch_info(&epoch_id)?;
+                Ok(serialize_entity(&*epoch_info))
+            }
+            EntityQuery::TransactionByHash { transaction_hash } => {
+                let transaction = self
+                    .store
+                    .get_ser::<SignedTransaction>(
+                        DBCol::Transactions,
+                        &transaction_hash.try_to_vec().unwrap(),
+                    )?
+                    .ok_or_else(|| anyhow!("Transaction not found"))?;
+                Ok(serialize_entity(&SignedTransactionView::from(transaction)))
+            }
+            EntityQuery::ReceiptById { receipt_id } => {
+                let receipt = self
+                    .store
+                    .get_ser::<Receipt>(DBCol::Receipts, &receipt_id.try_to_vec().unwrap())?
+                    .ok_or_else(|| anyhow!("Receipt not found"))?;
+                Ok(serialize_entity(&ReceiptView::from(receipt)))
+            }
+            EntityQuery::OutcomeByTransactionHash { transaction_hash: outcome_id }
+            | EntityQuery::OutcomeByReceiptId { receipt_id: outcome_id } => {
+                let (_, outcome) = self
+                    .store
+                    .iter_prefix_ser::<ExecutionOutcomeWithProof>(
+                        DBCol::TransactionResultForBlock,
+                        &outcome_id.try_to_vec().unwrap(),
+                    )
+                    .next()
+                    .ok_or_else(|| anyhow!("Outcome not found"))??;
+                Ok(serialize_entity(&ExecutionOutcomeView::from(outcome.outcome)))
```

Comment:
It could be more informative to display `ExecutionMetadata` because its gas profile is stored in more reasonable way. Unfortunately it doesn't implement `Serialize` :(

Reply:
Hmm, it seems that ExecutionOutcomeView does contain an ExecutionMetadataView which contains a flattened list from the data in ExecutionMetadata.

Reply:
Oh, I was under impression that it always uses legacy V2 format but it is not the case. Great!